config/main.py

```diff
@@ -1227,6 +1227,91 @@ def validate_patch(patch):
     except Exception as e:
         raise GenericConfigUpdaterError(f"Validate json patch: {patch} failed due to:{e}")
 
+
+def multiasic_validate_single_file(filename):
+    ns_list = [DEFAULT_NAMESPACE, *multi_asic.get_namespace_list()]
+    file_input = read_json_file(filename)
+    file_ns_list = [DEFAULT_NAMESPACE if key == HOST_NAMESPACE else key for key in file_input]
+    if set(ns_list) != set(file_ns_list):
+        click.echo("Input file {} must contain all asics config".format(filename))
+        raise click.Abort()
+
+
+def load_sysinfo_if_missing(asic_config):
+    platform = asic_config.get("DEVICE_METADATA", {}).\
+        get(HOST_NAMESPACE, {}).get("platform")
+    mac = asic_config.get("DEVICE_METADATA", {}).\
+        get(HOST_NAMESPACE, {}).get("mac")
+    if not platform or not mac:
+        log.log_warning("Input file does't have platform or mac. platform: {}, mac: {}"
+                        .format(None if platform is None else platform, None if mac is None else mac))
+        return True
+    return False
+
+
+def flush_configdb(namespace=DEFAULT_NAMESPACE):
+    if namespace is DEFAULT_NAMESPACE:
+        config_db = ConfigDBConnector()
+    else:
+        config_db = ConfigDBConnector(use_unix_socket_path=True, namespace=namespace)
```

Comment:
>namespace=namespace [](http://example.com/codeflow?start=65&length=19)

if namespace is empty, it will return host configdb, right?

Reply:
Right.

Reply:
This part of code was copied from the original. Seems one more difference for use_unix_socket_path. The default value is False. Not sure any historic reason on this.